src/useDynamicAttribute/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react'
+
+type Enrichment<T, TKey extends string, TSet extends string, TValue> =
+  & Record<TKey, TValue>
+  & Record<TSet, (value: TValue, data: (T & Enrichment<T, TKey, TSet, TValue>)[]) => void>
+
+export type Enriched<T, TKey extends string, TSet extends string, TValue> = T & Enrichment<T, TKey, TSet, TValue>
+
+export function useDynamicAttribute<
+  T extends {},
```

Comment:
Requirement for list object is that it always posesses a key property with a string value - You can find a type in utils/index called MapItem but perhaps we should rename it?